### DISCUSSION TITLE
Issue on updating a user with a relation

### DISCUSSION THREAD

**Discussion by carbogninalberto** [pocketbase/pocketbase#5258] [Q&A] [answered]
Hi there, 
Maybe there is something wrong I am doing but I have the following issue:

When I create a new `users` record I also create a new `subscriptions` record.
After that I update the `users` freshly created record with the relation.

The problem is that in the local environment, it's working fine but in prod is not working, the field remains empty.

This is the code:
```javascript
const subscription = await locals.adminPb.collection('subscriptions').create(data);

const user = await locals.adminPb.collection('users').update(record.id, {
subscription: subscription.id
});
```

I am guessing the transaction is not yet committed when I am trying to update the record, because the following code "fixed" the issue:

```javascript
const subscription = await locals.adminPb.collection('subscriptions').create(data);
await new Promise(resolve => setTimeout(resolve, 2000));
const user = await locals.adminPb.collection('users').update(record.id, {
subscription: subscription.id
});
```

Is there something I am missing?

Thank you

**Comment by ganigeorgiev** [comment]
What is the issue here? Is there any error thrown?

I don't know what is the problem you are experiencing but I don't think it is PocketBase related (especially if you are not able to reproduce it locally).

> I am guessing the transaction is not yet committed ...

No. If you are using `await` for the 2 API calls in a single function block then the operations are guaranteed to happen one after another and the `setTimeout` call shouldn't be needed. There is something else going on in your environment but without some minimal reproducible steps there is nothing much I can help with.

**Reply by ganigeorgiev** [reply]
Make sure that you don't have somewhere else that calls `locals.adminPb.collection('users').update` that could overwrite the `subscription` field.

**Reply by carbogninalberto** [reply]
Thank you for the reply and the clarification about the fact that the operation is guaranteed to happen in a sequential order.

I checked but in the signup endpoint is the only place I am updating.

Moreover, the pocketbase I am currently using in this stage (just testing out) for dev it's the same of dev, the only difference is that on "prod" I am running the sveltekit in the same server, in dev in my local machine.

Here the fullcode:

```javascript
import {generateUsername, serializeNonPOJOs} from '$lib/utils.ts';
import { redirect } from '@sveltejs/kit';

export const actions = {
	signup: async ({ locals, request }) => {
		const body = Object.fromEntries(await request.formData());
		try {
			let username = generateUsername(body.name.split(' ').join('')).toLocaleLowerCase();

			const record = await locals.adminPb.collection('users').create({
				username,
				...body
			});

			// send verification email
			await locals.adminPb.collection('users').requestVerification(body.email);

			const plan = await locals.adminPb.collection('plans').getFirstListItem('name="Free"');

			const data = {
				"user": record.id,
				"plan": plan.id,
				"remaining_quota": plan.monthly_quota,
				"type": "monthly",
				"data": {},
				"valid_until": "2100-01-01 00:00:00.000Z"
			};
			const subscription = await locals.adminPb.collection('subscriptions').create(data);
			console.log("subscription:", subscription);
			const user = await locals.adminPb.collection('users').update(record.id, {
				subscription: subscription.id
			});
			console.log('user:', user);
			await locals.pb.collection('users').authWithPassword(body.email, body.password, {expand: 'subscription,subscription.plan'});
			locals.user = serializeNonPOJOs(locals.pb.authStore.model);
		} catch (err) {
			console.error("Failed for lead: ", body?.email, body.name);
			console.error(JSON.stringify(err));
			return {
				error: err.message,
				...body,
				responseData: err.response.data
			}
		}

		redirect(302, '/');
	}
};

```

Moreover, the console log seems ok, it outputs the correct updated user object.
What I noticed is that it take one second more in the local env.

`adminPB` is a PocketBase instance for the admin user
`pb` is a PocketBase of the logged user.

I create this in the sever hook.

I noticed also that if I add the timer before the subscriptions record creation, it also works fine.

**Reply by ganigeorgiev** [reply, marked as answer]
I don't think I'll be able to help you but once again the response of the create/update API calls guarantee that the result has been written in the database before returning it to the user (unless you have custom pb_hooks that intercept/change this).

If you think that this is a PocketBase issue, try to create a **minimal** reproducible repo (showing also how the JS SDK instances are initialized) and I'll try to have a look at it but the above code sample is not enough to understand or reproduce your problem.

_Side-note: you don't need `serializeNonPOJOs` because the JS SDK operates with plain JS objects since v0.17+._

**Reply by carbogninalberto** [reply]
At this point it has to be an issue in my server/svelte setup. 
I will try to figure it out (I don't have custom pb_hooks yet).
Thank you Gani for the side note (and for this project).

**Reply by ganigeorgiev** [reply]
Also you usually shouldn't need to have both user->subscription and subscription->user relation fields and if it is not required for your use case I'd recommend to keep only one to avoid getting out of sync.

If the reason for the double relation is that you want to use it with `expand`, then you can try [back-relations](https://pocketbase.io/docs/working-with-relations/#back-relations) and do something like (assuming that relation will be in the `subscriptions` collection):

```js
await locals.pb.collection('users').authWithPassword(body.email, body.password, {
    expand: 'subscriptions_via_user.plan'
});
```
